packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import { getNextElement, focusAsync } from './focusUtilities'
+
+import { AutoFocusZoneProps } from './AutoFocusZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+
+/** AutoFocusZone is used to focus the first element inside it's children */
+export class AutoFocusZone extends React.Component<AutoFocusZoneProps, {}> {
+  private _root: { current: HTMLElement | null } = { current: null }
+  private _previouslyFocusedElementOutsideTrapZone: HTMLElement
+  private _previouslyFocusedElementInTrapZone?: HTMLElement
+
+  private createRef = elem => (this._root.current = ReactDOM.findDOMNode(elem) as HTMLElement)
+
+  static propTypes = {
+    as: customPropTypes.as,
+    className: PropTypes.string,
+    ariaLabelledBy: PropTypes.string,
+    firstFocusableSelector: PropTypes.string,
+    disableFirstFocus: PropTypes.bool,
+    focusPreviouslyFocusedInnerElement: PropTypes.bool,
+  }
+
+  static defaultProps: AutoFocusZoneProps = {
+    as: 'div',
```

Comment:
Doesn't make any sense, `as` defaults to `div`

Reply:
This is not `UIComponent`, that's why we have this as part of `defaultProps`.

Reply:
https://github.com/stardust-ui/react/blob/master/packages/react/src/lib/getElementType.tsx#L41
It handled in `getElementType()`

Reply:
Yep, I forgot we are directly using it. Thanks!